### ISSUE TITLE
config parsing can break on multiline with multiple quotes

### ISSUE THREAD

**Issue by ehuss** [libgit2/libgit2#5459]
### Reproduction steps
Create a `.gitconfig` with the following content and attempt to load it with libgit2:

```
[alias]
    foo = !ls  "x" \
        ls  "# comment2" \
        $HOME
```

### Expected behavior
libgit2 should be able to parse this into:
`!ls  x         ls  # comment2         $HOME`

### Actual behavior
Fails with `invalid configuration key`

### Version of libgit2 (release number or SHA1)
be36db28f64de8ed5ca9870708a56b6668c6c482

### Notes
I *think* the crux of the issue is [this line](https://github.com/libgit2/libgit2/blob/9a1024465cf3650d513d3e997435683fcf518e2b/src/config_parse.c#L354).  AIUI, when it is parsing the second line `in_quotes = 2` because `parse_variable` passes the quote count.  `!!in_quotes` is a value of 1 which means it thinks it is still within quotes.  A change roughly like this seems like a start (although the `in_quotes` could probably be renamed to `quote_count` consistently throughout instead of trying to use `%2` in various places):

```diff
diff --git a/src/config_parse.c b/src/config_parse.c
index 48ad1164f..8c0efbbbc 100644
--- a/src/config_parse.c
+++ b/src/config_parse.c
@@ -351,7 +351,7 @@ static int parse_multiline_variable(git_config_parser *reader, git_buf *value, i
                }

                /* If it was just a comment, pretend it didn't exist */
-               quote_count = strip_comments(line, !!in_quotes);
+               quote_count = strip_comments(line, in_quotes);
                if (line[0] == '\0')
                        goto next;
```

Reported by a user which has this elaborate alias: https://github.com/laughinghan/dotfiles/blob/cd846f8069fb10fb25df072da50ec6d9f25f98e7/.gitconfig#L34

**Comment by csware** [comment]
This is still an issue with 1e987525b3bd86fc29e7ff67d973861efd22faf7 it's even worse: teh config file with a new line only consisting of "$HOME" cannot be parsed at all.

I compiled two new test cases for this issue: https://github.com/csware/libgit2/tree/config-multiline-parse

**Comment by ethomson** [comment]
Fixed via #5629 - thanks @csware!
